TheJumpCloud/support
# Merge 1.21.0 & Backup-JCOrg changes

## Issues
* [SA-2014](https://jumpcloud.atlassian.net/browse/SA-2014) - Backup-JCOrganization Permission Errors During Backup (Read-Only API Key)

## What does this solve?

When an admin with a read-only api key attempts to run backup-jcorganization they would have been met with an error message if they attempted to backup associations between commands -> systems, commands -> system_groups; software_apps -> systems, software_apps ->system_groups. This fix does not address the issue upstream with the api but in the PowerShell module itself. 

If a read only admin runs either:
`Backup-JCOrganization -Path ~/backup -Type command, system -Associations -Debug`
`Backup-JCOrganization -Path ~/backup -Type command, systemGroup -Associations -Debug`
`Backup-JCOrganization -Path ~/backup -Type softwareApps, system -Associations -Debug`
`Backup-JCOrganization -Path ~/backup -Type softwareApps, systemGroup -Associations -Debug`

The PowerShell Module should through an error. 

## Is there anything particularly tricky?

## How should this be tested?

If a read only admin runs either after importing this module:
`Backup-JCOrganization -Path ~/backup -Type command, system -Associations -Debug`
`Backup-JCOrganization -Path ~/backup -Type command, systemGroup -Associations -Debug`
`Backup-JCOrganization -Path ~/backup -Type softwareApps, system -Associations -Debug`
`Backup-JCOrganization -Path ~/backup -Type softwareApps, systemGroup -Associations -Debug`

The PowerShell Module should **not** throw an error. 


## Screenshots
